csharp-basic-topics/ValueAndReferenceTypes/Tests/Test.cs

```diff
@@ -0,0 +1,77 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ValueAndReferenceTypes;
+namespace Tests
+{
+    [TestClass]
+    public class Test
+    {
+        [TestMethod]
+        public void WhenValueType_ThenBothRectanglesHaveSeparateCopies()
+        {
+            var firstRectangle = new ValueTypeRectangle();
+            firstRectangle.length = 10;
+            firstRectangle.breadth = 10;
+
+            var secondRectangle = firstRectangle;
+
+            firstRectangle.length = 20;
+            firstRectangle.breadth = 20;
+
+            Assert.AreNotEqual(firstRectangle.Area(), secondRectangle.Area());
+        }
+
+        [TestMethod]
+        public void WhenReferenceType_ThenBothRectanglesHaveReferencesToSameAddress()
+        {
+            var firstRectangle = new ReferenceTypeRectangle();
+            firstRectangle.length = 10;
+            firstRectangle.breadth = 10;
+
+            var secondRectangle = firstRectangle;
+
+            firstRectangle.length = 20;
+            firstRectangle.breadth = 20;
+
+            Assert.AreEqual(firstRectangle.Area(), secondRectangle.Area());
+        }
+
+        [TestMethod]
+        public void WhenPassByValue_ThenCounterHaveDifferentCopies()
+        {
+            var pass = new PassByValueAndReferenceSample();
+            int counter = 100;
+            pass.PassByValue(counter);
+
+            Assert.AreEqual(100, counter);
+        }
+
+        [TestMethod]
+        public void WhenPassByReference_ThenCounterHaveReferenceToSameAddress()
+        {
+            var pass = new PassByValueAndReferenceSample();
+            int counter = 100;
+            pass.PassByReference(ref counter);
+
+            Assert.AreEqual(101, counter);
+        }
+
+        [TestMethod]
+        public void When_BoxingUnBoxing()
```

Comment:
All test should have the When_Then sections and the underscore should be used to separate those two sections not like it's used here.